File: 0020-valid_parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []

        paren_map = {
          '}': '{',
          ')': '(',
          ']': '['
        }
        
        for i in s:
          if i in paren_map:
            if  stack and stack[len(stack) - 1] == paren_map.get(i) :
              stack.pop()
            else:
              return False
          else:
            stack.append(i)
        
        return not stack

File: 0020-valid_parentheses.go
func isValid(s string) bool {
    mapParens := map[rune]rune {
        ')': '(',
        ']': '[',
        '}': '{',
    }

    var stack []rune

    for _, paren := range s {
        if open, exists := mapParens[paren]; exists {
            if len(stack) == 0 || stack[len(stack)-1] != open {
                return false
            }

            stack = stack[:len(stack)-1]
        } else {
            stack = append(stack, paren)
        }
    }

    return len(stack) == 0
}